### PR TITLE
[api] Add prioritized and deprioritized values to gas estimation

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Provide a prioritized and deprioritized value for gas estimation

### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4937)

### CODE REVIEW

**Review by igor-aptos — APPROVED** [review]
I am not sure how exactly the callers will use this, but the computation looks good to me

**Comment by gregnazario** [comment]
> I am not sure how exactly the callers will use this, but the computation looks good to me

Callers will use the three numbers, and be provided the 3 choices.  The upstream logic has to decide which choice to use.

This can be improved later with information about mempool or historical data

File: api/src/context.rs [hunk 1]
```diff
@@ -666,33 +679,97 @@ impl Context {
...
+
+            // Previous bucket could be the same as the minimum price (if too low)
+            let previous_bucket_price = self
+                .node_config
+                .mempool
+                .broadcast_buckets
+                .get(bucket_index.saturating_sub(1))
+                .copied()
+                .unwrap_or(min_gas_price);
+            // Next bucket could be the same as the current gas price, but we increase by 1 for the estimation
```
Comment (movekevin) [inline api/src/context.rs:741]: Why do we need to add 1? The bucket already has it (e.g. 151, 301, etc. https://github.com/aptos-labs/aptos-core/blob/main/config/src/config/mempool_config.rs#L47)

File: api/src/transactions.rs [hunk 2]
```diff
@@ -406,6 +406,9 @@ impl TransactionsApi {
         /// If set to true, the gas unit price in the transaction will be ignored
         /// and the estimated value will be used
         estimate_gas_unit_price: Query<Option<bool>>,
+        /// If set to true, the transaction will use a higher price than the original
+        /// estimate.
+        estimate_prioritized_gas_unit_price: Query<Option<bool>>,
```
Comment (movekevin) [inline api/src/transactions.rs:411]: nit: Why not name this use_prioritized_gas_unit_price?

**Review by movekevin — APPROVED** [review]

**Comment by banool** [comment]
Could you regenerate the docs and the TS SDK?

**Pull request merged**